boards/arm/nucleo_h743zi/nucleo_h743zi.dts

```diff
@@ -87,6 +87,7 @@
 
 &usart3 {
 	pinctrl-0 = <&usart3_tx_pd8 &usart3_rx_pd9>;
+	pinctrl-names = "default";
```

Comment:
As a simplification (when we'll have to deploy the solution), I wonder if that would be doable to make `pinctrl-names` non required if only pinctrl-0 is provided  (and then defaulted to "default").

Reply:
In case of Linux, if `pinctrl-names` is missing, the index is assigned as a state name, in this case `"0"`, see https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/drivers/pinctrl/devicetree.c#n239. DT tooling should be modified to support this case, and we should probably pre-define some states (`PINCTRL_STATE_{0..N}`) so that drivers working this way do not need to do that themselves.

I think that if the driver uses standard states (`default`, `sleep`...) it is better to make them explicit in DT. It is easier to understand what `PINCTRL_STATE_DEFAULT` means when looking at driver+DT sources.